AirCasting/Services/DownloadMeasurmentsService.swift

```diff
@@ -38,12 +38,13 @@ class DownloadMeasurementsService: ObservableObject {
                 }
             } receiveValue: { [weak self] (fixedMeasurementOutput) in
                 guard let self = self else { return }
-                
-                // Fetch session by id from Core Data
-                let session: Session = self.context.newOrExisting(uuid: fixedMeasurementOutput.uuid)
-                UpdateSessionParamsService().updateSessionsParams(session: session, output: fixedMeasurementOutput)
-                
-                try! self.context.save()
+                DispatchQueue.main.async {
```

Comment:
Tu nie będzie tego samego problemu z kontekstem?

Reply:
I'm not sure if it's a similar case to that previous issue but I just want to make sure

Reply:
This thread change to main was done before inside the `getFixedMeasurement` so this is the code that moves back to the main thread so the `UpdateSessionParamsService().updateSessionsParams` that currently is using the core data `viewContext` will work. Final version of the code should use a child context of the `viewContext` into which the database update should be made.